pyaedt/modules/PostProcessor.py

```diff
@@ -1449,218 +1683,213 @@ def get_far_field_data(self, expression="GainTotal", setup_sweep_name='', domain
-            ``"dB(S(1,1))"``.
-        setup_sweep_name : str, optional
-            Name of the setup for computing the report. The
-            default is ``""``, in which case the nominal sweep is
-            used.
-        domain : str or list, optional
-            Context type. The options are ``"Sweep"`` or
-            ``"Time"``. The default is ``"Sweep".``
-        families_dict : dict, optional
-            Dictionary of all families including the primary
-            sweep. The default is ``{"Freq": ["All"]}``.
-        report_input_type :  str
-            Type of input data for the report.
-
-        Returns
-        -------
-        :class:`pyaedt.modules.PostProcessor.SolutionData`
-
-        Examples
-        --------
-        Generate a report with the default sweep and default variation.
-
-        >>> hfss = HFSS()
-        >>> hfss.post.get_report_data("S(1,1)")
-
-        >>> m3d = Maxwell3D()
-        >>> m3d.post.get_report_data("SurfaceLoss")   # Eddy Current examples
-        >>> m3d.post.get_report_data("Wind(LoadA,LaodA)")    # TransientAnalsysis
 
-        """
-        if self.post_solution_type in ["3DLayout", "NexximLNA", "NexximTransient"]:
-            if domain == "Sweep":
-                did = 3
-            else:
-                did = 1
-            ctxt = ["NAME:Context", "SimValueContext:=",
-                    [did, 0, 2, 0, False, False, -1, 1, 0, 1, 1, "", 0, 0, "IDIID", False, "1"]]
-        elif type(domain) is list:
-            ctxt = domain
-        else:
-            ctxt = ["Domain:=", domain]
 
-        if type(expression) is not list:
-            expression = [expression]
-        if not setup_sweep_name:
-            setup_sweep_name = self._parent.nominal_sweep
+class CircuitPostProcessor(PostProcessorCommon, object):
+    """Manages the main AEDT Nexxim postprocessing functions.
 
 
-        if not report_input_type:
-            report_input_type = report_type[self.post_solution_type]
+    .. note::
+   Some functionalities are available only when AEDT is running in the graphical mode.
 
+    Parameters
+    ----------
+    parent:
+        Inherited parent object. The parent object must provide the members
+        `_modeler`, `_desktop`, `_odesign`, and `_messenger`.
 
-        if families_dict is None:
-            families_dict = {"Freq": ["All"]}
+    """
 
-        solution_data = self.get_solution_data_per_variation(report_input_type, setup_sweep_name, ctxt, families_dict, expression)
+    def __init__(self, parent):
+        PostProcessorCommon.__init__(self, parent)
 
-        if not solution_data:
-            print("No Data Available. Check inputs")
-            return False
-        return solution_data
+    def create_ami_initial_response_plot(self, setupname, ami_name, variation_list_w_value, plot_type="Rectangular Plot", plot_initial_response=True, plot_intermediate_response=False, plot_final_response=False,plotname=None):
```

Comment:
Let's also consider changing `setupname` and `plotname` to use camel case.


```suggestion
    def create_ami_initial_response_plot(self, setupname, ami_name, variation_list_w_value, 
                                         plot_type="Rectangular Plot", plot_initial_response=True,
                                         plot_intermediate_response=False, plot_final_response=False, plotname=None):
```